lib/jsonapi/view.ex

```diff
@@ -91,7 +91,19 @@ defmodule JSONAPI.View do
       end
 
       def attributes(data, conn) do
-        visible_fields(data)
+        visible_fields = fields() -- hidden()
+
+        Enum.reduce(visible_fields, %{}, fn field, intermediate_map ->
+          try do
+            merge(intermediate_map, field, apply(__MODULE__, field, [data, conn]))
```

Comment:
Can we do something like this to DRY things up a bit?

```elixir
value =
  try do
    apply(__MODULE__, field, [data, conn])
  rescue
    _e -> Map.get(data, field)
  end

Map.put(intermediate_map, field, value)
```

Reply:
ah much better :100: